lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,565 @@
+        tokenizers_parallelism: bool = True,
+        num_workers: int = 0,
+    ):
+        super().__init__()
+        self.model_name_or_path = model_name_or_path
+        self.task_name = task_name if task_name in TASK_NUM_LABELS.keys() else DEFAULT_TASK
+        self.max_seq_length = max_seq_length
+        self.train_batch_size = train_batch_size
+        self.eval_batch_size = eval_batch_size
+        self.tokenizers_parallelism = tokenizers_parallelism
+        self.dataloader_kwargs = {"num_workers": num_workers, "pin_memory": pin_memory}
+        self.text_fields = self.task_text_field_map[self.task_name]
+        self.num_labels = TASK_NUM_LABELS[self.task_name]
+        os.environ["TOKENIZERS_PARALLELISM"] = "true" if self.tokenizers_parallelism else "false"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True, local_files_only=False)
+        if prep_on_init:  # useful if one wants to load datasets as soon as the ``LightningDataModule`` is
+            # instantiated
+            self.prepare_data()
+            self.setup("fit")
+
+    def setup(self, stage):
+        self.dataset = datasets.load_dataset("super_glue", self.task_name)
+        for split in self.dataset.keys():
+            self.dataset[split] = self.dataset[split].map(
+                self.convert_to_features, batched=True, remove_columns=["label"]
+            )
+            self.columns = [c for c in self.dataset[split].column_names if c in self.loader_columns]
+            self.dataset[split].set_format(type="torch", columns=self.columns)
+
+        self.eval_splits = [x for x in self.dataset.keys() if "validation" in x]
+
+    def prepare_data(self):
+        # N.B. PL calls prepare_data from a single process (rank 0) so do not use it to assign
+        # state (e.g. self.x=y)
+        datasets.load_dataset("super_glue", self.task_name)
+
+    def train_dataloader(self):
+        return DataLoader(self.dataset["train"], batch_size=self.train_batch_size, **self.dataloader_kwargs)
+
+    def val_dataloader(self):
+        if len(self.eval_splits) == 1:
+            return DataLoader(
+                self.dataset["validation"],
+                batch_size=self.eval_batch_size,
+                **self.dataloader_kwargs,
+            )
+        elif len(self.eval_splits) > 1:
+            return [
+                DataLoader(self.dataset[x], batch_size=self.eval_batch_size, **self.dataloader_kwargs)
+                for x in self.eval_splits
+            ]
+
+    def test_dataloader(self):
+        if len(self.eval_splits) == 1:
+            return DataLoader(self.dataset["test"], batch_size=self.eval_batch_size, **self.dataloader_kwargs)
+        elif len(self.eval_splits) > 1:
+            return [
+                DataLoader(self.dataset[x], batch_size=self.eval_batch_size, **self.dataloader_kwargs)
+                for x in self.eval_splits
+            ]
+
+    def convert_to_features(self, example_batch):
```

Comment:
Should be private.

Reply:
Is it okay if I switch it to protected rather than private? I agree it shouldn't be public.

Reply:
Yes